src/spectro_inlets_quantification/sensitivity.py

```diff
@@ -218,13 +230,41 @@ def __len__(self) -> int:
         """Return the number of the SensitivityFactors in the object."""
         return len(self.sf_list)
 
-    def __add__(self, other: "SensitivityList") -> "SensitivityList":
-        """Return a `SensitivityList` with the sensitivity factors of itself and ``other``."""
+    def __add__(self, other: [SensitivityFactor or "SensitivityList"]) -> "SensitivityList":
+        """Return a `SensitivityList` with the sensitivity factors of self and ``other``.
+
+        You can add SensitivityFactor, CalPoint, SensitivityList, or Calibration to a
+        SensitivityList or Calibration.
+        If either self or other is a Calibration, the result will be a Calibration.
+        """
+        if issubclass(other.__class__, self.__class__):
+            # This ensures that SensitivityList + Calibration -> Calibration
+            return other + self
+
+        if isinstance(other, SensitivityFactor):
+            # Adding a SensitivityFactor (or CalPoint) to a SensitivityList (or Calibration)
+            # just makes a copy with the new SensitivityFactor appended to the
+            # SensitivityList's list.
+            obj_as_dict = self.as_dict()
+            if "cal_dicts" in obj_as_dict:
+                obj_as_dict["cal_list"] = obj_as_dict["cal_list"] + [other]
+            else:
+                obj_as_dict["sf_list"] = obj_as_dict["sf_list"] + [other]
+            return self.__class__(**obj_as_dict)
+
         if isinstance(other, SensitivityList):
+            # Adding a SensitivityList to a SensitivityList (or Calibration) makes a
+            # copy with the two lists of sensitivity factors appended.
             sf_list = self.sf_list + other.sf_list
-        else:
-            raise TypeError(f"can't add {other} to {self}")
-        return SensitivityList(sf_list)
+            cls = self.__class__
```

Comment:
Maybe a bit of nitpicking her, but here you give the class an alias, above you just use it from `self.__class__`, it seems a little random

Reply:
Now the alias `cls` is used both places. I like the habit of using an alias for this because sometimes the determination of the type to return is more involved.